magento/pwa-studio
# Fetching and displaying errors when adding product to cart

## Description
When adding a product to the cart, if there is an error (eg insufficient stock), the product fails to add to cart and there is no feedback to the user.

- Adding `user_errors` to graphql response when adding products to cart
- Adding graphql response errors to front-end errors

## Related Issue







Closes #3953 

## Acceptance





### Verification Stakeholders



### Specification

No special verification

## Verification Steps






#### Test scenario(s) for direct fix/feature

1. Go to a product page
2. Add 900 qty of the product to your cart
3. Try to repeat the action of adding 900 to your cart
4. There should be an error message displayed under the quantity saying there's insufficient stock

#### Test scenario(s) for any existing impacted features/areas

None

#### Test scenario(s) for any Magento Backend Supported Configurations

None



#### Is Browser/Device testing needed?

No




#### Any ad-hoc/edge case scenarios that need to be considered?

No



## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)

None




## Checklist
